src/hooks/api/index.ts

```diff
@@ -63,14 +63,17 @@ export function useAccountsMonthlyTotals(): SWRResponse<queries.MonthlyTotals> {
     fetcher(PriceDB.getTodayQuotes, '/api/prices/today'),
   );
 
-  const key = '/api/accounts/monthly-totals';
+  const key = (accounts && todayPrices) ? [
+    '/api/accounts/monthly-totals',
+    // We need to recompute when accounts and prices change
+    Object.keys(accounts),
+    Object.keys(todayPrices.keys),
+  ] : null;
```

Comment:
with previous approach, even when accounts and prices changed, the monthly-totals were not being recalculated. By having a different key every time we have new data we make sure the monthly totals are updated to the latest.